Journalist can add an image

rithreaksa opened the pull request:
(https://www.pivotaltracker.com/story/show/176031723)
### User Story
```
As a journalist
In order to make my articles more visually appealing
I would like to be able to upload and display images to my articles

```
## Changes proposed in this pull request:

* Writes a test for attaching an image
* Adds an image to fixture file
* Adds form input for attaching an image
* Adds functionality to send binary image data to API

## What I have learned working on this feature:

* Writes test for attaching an image
* Functionality to send binary image data to API

## Packages/Gems added

* cypress-file-upload

## Screenshots (Client)

<img width="1280" alt="Screen Shot 2021-01-09 at 2 29 23 PM" src="https://user-images.githubusercontent.com/62254746/104092878-1d442580-5287-11eb-811c-ef6c99b7ee90.png">

rithreaksa committed:
Writes a test for attaching an image
Adds an image to fixture file
Add filed input for attaching an image

rithreaksa committed:
Modifies 'data-cy' to 'name' in journalist can create an article

rithreaksa committed:
Modifies the name of input field of image

sealfury committed:
Removes console logging from auth module
Removes html breaks in create article form
begins image base64 encoding functionality

sealfury committed:
Adding encoded image to POST request params
Adding functionality for image preview and some styling

sealfury committed:
matches article POST req params to API naming conv

swardbladpaulo commented on src/modules/auth.js:
```diff
@@ -25,9 +25,7 @@ const performAuthentication = async (e, dispatch) => {
-  } 
-  catch (error) {
-    console.log(error);
+  } catch (error) {
```
Here you removed the cosole.log(error), is there any particulare reason for that?

rithreaksa replied:
It's because we have a dispatch function that set error message when error is caught, so we thought console.log is not needed anymore.

FlisAnn commented on cypress/integration/journalistCanCreateArticles.feature.js:
```diff
@@ -1,96 +1,98 @@
 
-  it("successfully", () => {
+  it('successfully', () => {
+    const title = 'Covid vaccine is found';
```
the test is changed to use a const for the title, why is that? Should it be changed for subtitle and body also for consistency?

sealfury replied:
Hmm good catch, will either def all typed inputs as constants or none of them

rithreaksa replied:
oh!! We will remove that const.. it was me who played around to test the image thingy. Thanks.

FlisAnn commented on cypress/integration/journalistCanCreateArticles.feature.js:
```diff
@@ -1,96 +1,98 @@
-        "Russia has distributed vaccines months ago"
+        'Russia has distributed vaccines months ago'
       );
+      cy.get('[name="file_input"]').attachFile('image.png');
```
Is it possible to have a proper image that shows correctly so we really can see that the image is found correctly?

sealfury replied:
That is a proper image, it’s saved in the fixture file. We also have an image preview function in the form input; the test shows this all happening

FlisAnn replied:
The image looks broken when running the test,  that's why we were wondering

rithreaksa replied:
ohh that's strange. I tried running on mine and it was okay. Thanks.

sealfury replied:
Must run only the successful test to see it displayed properly :)

swardbladpaulo commented on cypress/integration/journalistCanCreateArticles.feature.js:
```diff
@@ -1,96 +1,98 @@
-        "Russia has distributed vaccines months ago"
+        'Russia has distributed vaccines months ago'
       );
+      cy.get('[name="file_input"]').attachFile('image.png');
```
If the image is not there or broken, should it not be an alternative text there?

sealfury replied:
We’ll consider this- we felt that the image preview takes care of this, and I believe that unlike the other form fields, adding an image was meant to be an option for journalists rather than a requirement.

sealfury replied:
Removed the constants from the test

sealfury committed:
removing constants from test
rewriting error message to be more explicit
fixing naming conventions in login test

emtalen commented on cypress/integration/journalistCanCreateArticles.feature.js:
```diff
@@ -1,96 +1,97 @@
-        response: { message: "Something went wrong!" },
+        method: 'POST',
+        url: 'http://localhost:3000/api/articles',
+        response: { message: 'Please fill in all input fields!' },
```
Is this really the message that you are sending from the api?

sealfury replied:
It wasn’t previously, but the plan was to change it during the corresponding api user story

emtalen commented on src/modules/CreateArticle.js:
```diff
@@ -4,14 +4,27 @@ const CreateArticle = {
     e.preventDefault();
     let headers = JSON.parse(localStorage.getItem("J-tockAuth-Storage"));
+
+    const toBase64 = (file) =>
```
please make this a separate function, so it is not a part of the `create` function. I would suggest moving it to a separate module and import it here

sealfury replied:
Good call, we've done this now

sealfury committed:
Extracts toBase64 function to separate module

sealfury committed:
Fixing syntax typo in toBase64 function
